src/elements/content-sidebar/DocGenSidebar/DocGenSidebar.tsx

```diff
@@ -107,20 +114,23 @@ const DocGenSidebar = ({ getDocGenTags }: Props) => {
                     imageTree: tagsToJsonPaths(imageTags),
                 });
                 setHasError(false);
+                setIsLoading(false);
             } else {
                 setHasError(true);
+                setIsLoading(false);
             }
         } catch (error) {
             setHasError(true);
+            setIsLoading(false);
         }
-        setIsLoading(false);
-    };
-
+    }
+    
     React.useEffect(() => {
-        loadTags();
+        loadTags(10);
```

Comment:
Idea, add `loadTags` to dependency array and get rid of `react-hooks/exhaustive-deps`. And to make sure it doesn't get called many times wrap `loadTags` in a `React.useCallback`.